3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,75 @@
+// Global variables to store user data
+let userName = "";
+const todoList = [];
+let timer;
+
+
+function getReply(command) {
+  if (command.startsWith("Hello my name is ")) {
+    
+    const name = command.replace("Hello my name is ", "");
+    userName = name;
+    return `Nice to meet you ${userName}`;
+  } else if (command === "What is my name") {
+    if (userName) {
+      return `Your name is ${userName}`;
+    } else {
+      return "I don't know your name yet.";
+    }
+  } else if (command.startsWith("Add ")) {
+    if (command.includes("to my todo")) {
+      
+      const todoItem = command.replace("Add ", "").replace(" to my todo", "");
+      todoList.push(todoItem);
+      return `${todoItem} added to your todo`;
+    }
+  } else if (command.startsWith("Remove ")) {
+    if (command.includes("from my todo")) {
+      
+      const todoItem = command.replace("Remove ", "").replace(" from my todo", "");
+      const index = todoList.indexOf(todoItem);
+      if (index !== -1) {
+        todoList.splice(index, 1);
+        return `Removed ${todoItem} from your todo`;
+      } else {
+        return `${todoItem} was not found in your todo list`;
+      }
+    }
+  } else if (command === "What is on my todo?") {
+    if (todoList.length === 0) {
+      return "Your todo list is empty";
+    } else {
+      const todoItems = todoList.join(" and ");
+      const plural = todoList.length > 1 ? "s" : "";
+      return `You have ${todoList.length} todo${plural} - ${todoItems}`;
+    }
+  } else if (command === "What day is it today?") {
+    const currentDate = new Date();
+    const options = { year: "numeric", month: "long", day: "numeric" };
+    return currentDate.toLocaleDateString(undefined, options);
+  } else if (command.includes("Set a timer for")) {
+    
+    const time = parseInt(command.match(/\d+/)[0]);
+    timer = setTimeout(() => {
+      console.log("Timer done");
+    }, time * 60000); // Convert minutes to milliseconds
+    return `Timer set for ${time} minutes`;
+  } else if (command.includes("What is ")) {
+    
+    try {
+      const result = eval(command.replace("What is ", ""));
+      return result;
+    } catch (error) {
+      return "Sorry, I couldn't perform the calculation.";
+    }
+  } else {
+    return "I don't understand that command";
+  }
+}
+
+// Example usage
+console.log(getReply("Hello my name is Benjamin")); // "Nice to meet you Benjamin"
+console.log(getReply("What is my name?")); // "Your name is Benjamin"
```

Comment:
This will print out `Sorry, I couldn't perform the calculation.`. Try to find out what is wrong.